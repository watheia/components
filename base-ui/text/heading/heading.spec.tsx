import React from 'react';
import { render } from '@testing-library/react';

import {
  H1Example,
  H2Example,
  H3Example,
  H4Example,
  H5Example,
  H6Example,
} from './heading.composition';

it('should render correct text in Example', () => {
  const { getByText } = render(<H1Example />);
  const container = getByText('H1 Example');
  expect(container).toBeInstanceOf(HTMLElement);
  expect(container).toMatchSnapshot();
});

it('should render as h2 when defined', () => {
  const { getByText } = render(<H2Example />);
  const container = getByText('H2 Example');
  expect(container.tagName).toEqual('H2');
  expect(container).toMatchSnapshot();
});

it('should render as h3 when defined', () => {
  const { getByText } = render(<H3Example />);
  const container = getByText('H3 Example');
  expect(container.tagName).toEqual('H3');
  expect(container).toMatchSnapshot();
});

it('should render as h4 when defined', () => {
  const { getByText } = render(<H4Example />);
  const container = getByText('H4 Example');
  expect(container.tagName).toEqual('H4');
  expect(container).toMatchSnapshot();
});

it('should render as h5 when defined', () => {
  const { getByText } = render(<H5Example />);
  const container = getByText('H5 Example');
  expect(container.tagName).toEqual('H5');
  expect(container).toMatchSnapshot();
});

it('should render as h6 when defined', () => {
  const { getByText } = render(<H6Example />);
  const container = getByText('H6 Example');
  expect(container.tagName).toEqual('H6');
  expect(container).toMatchSnapshot();
});
