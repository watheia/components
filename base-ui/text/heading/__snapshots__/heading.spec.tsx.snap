// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as h2 when defined 1`] = `
<h2
  class="h2 lg"
  data-bit-id="waweb.base-ui/text/heading"
>
  H2 Example
</h2>
`;

exports[`should render as h3 when defined 1`] = `
<h3
  class="h3 md"
  data-bit-id="waweb.base-ui/text/heading"
>
  H3 Example
</h3>
`;

exports[`should render as h4 when defined 1`] = `
<h4
  class="h4 sm"
  data-bit-id="waweb.base-ui/text/heading"
>
  H4 Example
</h4>
`;

exports[`should render as h5 when defined 1`] = `
<h5
  class="h5 xs"
  data-bit-id="waweb.base-ui/text/heading"
>
  H5 Example
</h5>
`;

exports[`should render as h6 when defined 1`] = `
<h6
  class="h6 xxs"
  data-bit-id="waweb.base-ui/text/heading"
>
  H6 Example
</h6>
`;

exports[`should render correct text in Example 1`] = `
<h1
  class="h1 lg"
  data-bit-id="waweb.base-ui/text/heading"
>
  H1 Example
</h1>
`;
